src/install/install.zig

```diff
@@ -4415,9 +4415,7 @@ pub const PackageManager = struct {
                 if (update.e_string) |e_string| {
                     e_string.data = switch (update.resolution.tag) {
                         .npm => if (update.version.tag == .npm and update.version.value.npm.version.input.len == 0)
-                            std.fmt.allocPrint(allocator, "^{}", .{
```

Comment:
If you do

```
bun add foo
```

it still writes

`foo: "^1.2.3"`


 right?

Reply:
It should only write `latest` when the user explicitly passes `latest`, like 
```
bun add foo@latest
```

Reply:
Turns out that's the wrong behaviour &minus; as illustrated in the test this PR, once you do that you ended up not picking the `latest` tagged version which should be the default when no versions is specified:

https://docs.npmjs.com/cli/v9/commands/npm-install
> npm install [<@scope>/]<name>:
> 
> Do a <name>@<tag> install, where <tag> is the "tag" [config](https://docs.npmjs.com/cli/v9/using-npm/config#tag). (See config. The config's default value is latest.)
> 
> In most cases, this will install the version of the modules tagged as latest on the npm registry.

Reply:
<img width="453" alt="image" src="https://user-images.githubusercontent.com/709451/217680981-91098e1c-e07e-4231-a85d-1747607a4415.png">

Reply:
ah it also keeps it when using the explicit tag 
<img width="417" alt="image" src="https://user-images.githubusercontent.com/709451/217681099-d7511a24-d71d-4fdb-aa2d-e1543aae5831.png">

Reply:
Hmm, that means we have a dilemma here &minus; `npm` probably uses `package-lock.json` to freeze out the version thus is free to write that out without worrying about subsequent `npm install` picking the wrong version.

Reply:
let's match npm behavior